indico/MaKaC/services/implementation/contribution.py

```diff
@@ -55,13 +53,8 @@ def _checkParams( self ):
         if self._conf == None:
             raise Exception("Conference id not specified.")
 
-        try:
-            self._target = self._contribution = self._conf.getContributionById(self._params["contribution"])
-        except:
-            try:
-                self._target = self._contribution = self._conf.getContributionById(self._params["contribId"])
-            except:
-                raise ServiceError("ERR-C0", "Invalid contribution id.")
+        contrib_id = int(self._params.get('contribId', self._params.get('contribution')))
+        self._target = self._contribution = self.contrib = Contribution.get(contrib_id)
```

Comment:
`get_one()` to raise an exception (that's not as misleading as the one right below) if there's no contribution with that ID?

Also, assert `self.contrib.event_id == int(self._conf.id)`